File: ios/Plugin/W3aCustomPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(W3aCustomPlugin, "W3aCustom",
           CAP_PLUGIN_METHOD(echo, CAPPluginReturnPromise);
)

File: ios/Plugin/W3aCustomPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(W3aCustomPlugin)
public class W3aCustomPlugin: CAPPlugin {
    private let implementation = W3aCustom()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
